AdyenCheckout/CheckoutComponentBuilder.swift

```diff
@@ -32,37 +54,35 @@ internal enum CheckoutComponentBuilder {
     ) -> ActionComponent {
         fatalError()
     }
-}
-
-extension CheckoutConfiguration {
-    func componentConfiguration(for paymentMethod: PaymentMethod) -> CheckoutComponentConfiguration? {
-        configurations[.payment(paymentMethod.type)]
-    }
     
-//    func componentConfiguration(for action: Action) -> CheckoutComponentConfiguration? {
-//        configurations[.action(...)]
-//    }
-}
-
-// TODO: testing different ways of creating the component
-// we can go back to payment component builder or choose another way as well
-extension BLIKPaymentMethod {
-    
-    func buildComponent(with configuration: CheckoutConfiguration) -> BLIKComponent {
-        var blikConfiguration: BLIKComponentConfiguration
-        if let configuration = configuration.componentConfiguration(for: self) as? BLIKComponentConfiguration {
-            blikConfiguration = configuration
-        } else {
-            blikConfiguration = .init()
-        }
-        // TODO: find a better place to carry over global settings to individual configs
-        blikConfiguration.showsSubmitButton = configuration.showsSubmitButton
-        blikConfiguration.theme = configuration.theme
-        let component = BLIKComponent(
-            paymentMethod: self,
+    /// Creates a component using the provided factory for standard payment methods.
+    ///
+    /// This works for all components whose configurations conform to
+    /// `CheckoutComponentConfiguration`.
+    ///
+    /// - Parameters:
+    ///   - factory: The factory to use for component creation.
+    ///   - paymentMethod: The payment method to create a component for.
+    ///   - configuration: The checkout configuration.
+    /// - Returns: A configured payment component.
+    private static func createComponent<Factory: PaymentComponentFactory>(
+        using factory: Factory,
+        paymentMethod: Factory.Method,
+        configuration: CheckoutConfiguration
+    ) -> PaymentComponent where Factory.Configuration: CheckoutComponentConfiguration {
+        
+        var componentConfiguration = configuration.configuration(
+            for: paymentMethod,
+            defaultValue: factory.defaultConfiguration()
+        )
+        
+        componentConfiguration.showsSubmitButton = configuration.showsSubmitButton
```

Comment:
Why do we know about configuration properties and adjust them here? Shouldn't it be enclosed in the configuration factory?